terraform/shared/shared.tf.json.template.py

```diff
@@ -871,23 +870,32 @@ def paren(s: str) -> str:
                 'name': aws.monitoring_topic_name
             }
         },
-        'aws_sns_topic_subscription': {
-            'monitoring': {
-                'topic_arn': '${aws_sns_topic.monitoring.arn}',
-                # The `email` protocol is only partially supported. Since
-                # Terraform cannot confirm or delete pending subscriptions
-                # (see link below), we use a separate script for this purpose.
-                # https://registry.terraform.io/providers/hashicorp/aws/4.3.0/docs/resources/sns_topic_subscription#protocol-support
-                'protocol': 'email',
-                'endpoint': config.monitoring_email,
-                'provisioner': {
-                    'local-exec': {
-                        'command': ' '.join(map(shlex.quote, [
-                            'python',
-                            config.project_root + '/scripts/confirm_sns_subscription.py'
-                        ]))
-                    }
-                }
+        'aws_ses_domain_identity': {
+            'notify': {
+                'domain': config.domain_name
+            }
+        },
+        'aws_ses_identity_policy': {
+            'notify': {
+                'identity': '${aws_ses_domain_identity.notify.arn}',
+                'name': config.qualified_resource_name('notify'),
+                'policy': json.dumps({
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            'Effect': 'Allow',
+                            'Principal': {
+                                'AWS': f'arn:aws:sts::{aws.account}:assumed-role/{config.indexer_name}/'
+                                       + config.indexer_function_name('notify')
```

Comment:
I think we should use either f-strings or the `+` operator but not both

```suggestion
                                'AWS': f'arn:aws:sts::{aws.account}:assumed-role/{config.indexer_name}/'
                                       f'{config.indexer_function_name('notify')}'
```